File: _1.20.1/src/main/java/cn/enaium/noexpensive/callback/impl/AnvilTakeOutputCallbackImpl.java
package cn.enaium.noexpensive.callback.impl;

import cn.enaium.noexpensive.callback.AnvilTakeOutputCallback;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NbtCompound;
import net.minecraft.nbt.NbtElement;
import net.minecraft.nbt.NbtList;

/**
 * @author Enaium
 */
public class AnvilTakeOutputCallbackImpl implements AnvilTakeOutputCallback {
    @Override
    public void interact(ItemStack output) {
        NbtCompound nbtCompound = output.getOrCreateSubNbt(ItemStack.DISPLAY_KEY);
        final NbtList list = nbtCompound.getList(ItemStack.LORE_KEY, NbtElement.STRING_TYPE);
        for (int i = 0; i < list.size(); i++) {
            if (list.getString(i).contains("container.repair.cost")) {
                list.remove(i);
                break;
            }
        }
    }
}

File: modern/_1.19.4/src/main/kotlin/cn/enaium/noexpensive/callback/impl/AnvilTakeOutputCallbackImpl.kt
package cn.enaium.noexpensive.callback.impl

import cn.enaium.noexpensive.callback.AnvilTakeOutputCallback
import net.minecraft.client.MinecraftClient
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.item.ItemStack
import net.minecraft.nbt.NbtElement

/**
 * @author Enaium
 */
class AnvilTakeOutputCallbackImpl : AnvilTakeOutputCallback {
    override fun interact(output: ItemStack, player: PlayerEntity) {
        MinecraftClient.getInstance().player == player || player.abilities.creativeMode && return
        val nbtCompound = output.getOrCreateSubNbt(ItemStack.DISPLAY_KEY)
        val list = nbtCompound.getList(ItemStack.LORE_KEY, NbtElement.STRING_TYPE.toInt())
        for (i in list.indices) {
            if (list.getString(i).contains("container.repair.cost")) {
                list.removeAt(i)
                break
            }
        }
    }
}
